src/main/java/graphql/schema/transform/FieldVisibilitySchemaTransformation.java

```diff
@@ -250,12 +253,21 @@ public TraversalControl visitGraphQLType(GraphQLSchemaElement node,
         }
     }
 
-    private List<GraphQLObjectType> getRootTypes(GraphQLSchema schema) {
+    private List<GraphQLSchemaElement> getRootTypes(GraphQLSchema schema) {
+        return ImmutableList.<GraphQLSchemaElement>builder()
+                .addAll(getOperationTypes(schema))
+                // Include directive definitions as roots, since they won't be removed in the filtering process.
+                // Some types (enums, input types, etc.) might be reachable only by directive definitions (and
+                // not by other types or fields).
+                .addAll(schema.getDirectives())
```

Comment:
should we have schema.getAdditionalTypes here as well

Reply:
I guess this ends up removing types that are no linked (if I recall right) so perhap no

Reply:
I think the opposite would happen: types that became unreachable after fields were filtered out would still show up in the final schema, because essentially all types would be root types and thus reachable.